lib/geoengineer/resource.rb

```diff
@@ -155,19 +157,27 @@ def build_individual_remote_resource
   end
 
   def matched_remote_resource
-    self.class.fetch_remote_resources.select { |r| r._geo_id == _geo_id }
+    self.class.fetch_remote_resources(fetch_provider).select { |r| r._geo_id == _geo_id }
+  end
+
+  def fetch_provider
+    environment&.find_provider(provider)
```

Comment:
two types of provider, the provider string on a resource, and the provider object with all the attributes. This takes the provider string and returns the object

```
provider = env.provider("aws") {
  self["alias"] = "region_2"
  region ENV["REGION_2"]
}

env.resource("aws_resource", "resource") {
  provider    "aws.region_2"
}
```